### PR TITLE
varatube refactoring

### PR DESCRIPTION
**Pull request by LouiseMedova**

### CODE REVIEW

**Review by MedovTimur — APPROVED** [review]

File: contracts/varatube/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    /// Add subscriber.
+    pub fn add_subscriber(&mut self, subscriber: &ActorId, data: SubscriberData) {
+        self.subscribers.insert(*subscriber, data);
+    }
+
+    /// Add pending subscription.
+    ///
+    /// Inserting `data` is actually currency id and subscription period.
+    pub fn add_pending_subscriber(&mut self, subscriber: &ActorId, data: (ActorId, Period)) {
+        let (currency_id, period) = data;
```
Comment (shamilsan) [inline contracts/varatube/src/utils.rs:15]: We can destruct params in arg list.
```suggestion
    pub fn add_pending_subscriber(&mut self, subscriber: &ActorId, (currency_id, period): (ActorId, Period)) {
```

File: contracts/varatube/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        );
+    }
+
+    /// Ger subscriber.
+    pub fn get_subscriber(&self, subscriber: &ActorId) -> Result<SubscriberData, Error> {
+        if let Some(subscriber_data) = self.subscribers.get(subscriber) {
+            Ok(*subscriber_data)
+        } else {
+            Err(Error::AccountDoesNotExist)
+        }
```
Comment (shamilsan) [inline contracts/varatube/src/utils.rs:33]: Can we use `Option::ok_or` here?

```suggestion
        self.subscribers.get(subscriber)[.copied()].ok_or(Error::AccountDoesNotExist)
```

File: contracts/varatube/src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    pub fn get_price(&self, currency_id: &ActorId) -> Result<Price, Error> {
+        if let Some(price) = self.currencies.get(currency_id) {
+            Ok(*price)
+        } else {
+            Err(Error::UnregisteredPaymentMethod)
+        }
+    }
+
+    pub fn check_if_subscriber_doesnt_exist(&self, subscriber: &ActorId) -> Result<(), Error> {
+        if self.subscribers.get(subscriber).is_some() {
```
Comment (shamilsan) [inline contracts/varatube/src/utils.rs:51]:
```suggestion
        if self.subscribers.contains_key(subscriber) {
```

File: contracts/varatube/src/utils.rs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                subscriber: *subsciber,
+            },
+            gas_limit,
+            0,
+            delay,
+        )
+        .is_err()
+        {
+            return Err(Error::ErrorDuringSendingDelayedMsg);
+        }
```
Comment (shamilsan) [inline contracts/varatube/src/utils.rs:117]:
```suggestion
        .map_err(|_| Error::ErrorDuringSendingDelayedMsg)?;
```

File: contracts/varatube/tests/utils.rs [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    fn get_subscriber_data(&self, subscriber: u64) -> Option<SubscriberData> {
+        let state: StateReply = self
+            .read_state(StateQuery::GetSubscriber(subscriber.into()))
+            .expect("Unable to read varatube state");
+        if let StateReply::SubscriberData(subscriber_data) = state {
+            subscriber_data
+        } else {
+            gstd::panic!("Wrong received reply");
```
Comment (shamilsan) [inline contracts/varatube/tests/utils.rs:185]: Why not `std::panic`?
```suggestion
            panic!("Wrong received reply");
```

**Review by shamilsan — APPROVED** [review]

**Pull request merged**